.github/workflows/bump-opbeans-node.yml

```diff
@@ -0,0 +1,39 @@
+---
+## Workflow to periodically check if there is an available newer APM agent version, e.g.
+## "1.2.3". If so, then update to it and tag this repo with that version, e.g.
+## "v1.2.3".
+name: Bump opbeans node
+
+on:
+  schedule:
+    - cron:  '0 20 * * 6'
+
+jobs:
+  bump:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14
+          cache: 'npm'
+      - name: Get available agent update version
+        run: |
+          CURR=$(jq '.dependencies."elastic-apm-node".version' package-lock.json)
+          LATEST=$(npm info elastic-apm-node@latest version)
+          if [[ "${CURR}" != "${LATEST}" ]]; then
+            echo "AGENT_VERSION=${LATEST}" >> $GITHUB_ENV
+          fi
+      - name: Update version
+        if: ${{ env.AGENT_VERSION }}
+        run: |
+          npm install --ignore-scripts elastic-apm-node@${{ env.AGENT_VERSION }}
+          sed -ibck "s#\(org.label-schema.version=\)\(\".*\"\)\(.*\)#\1\"${{ env.AGENT_VERSION }}\"\3#g" Dockerfile
+          rm -f Dockerfilebck
```

Comment:
Alright, will try how updatecli could be utilized here.

Reply:
```yaml
name: NPM Elastic resource example

scms:
  githubConfig:
    kind: "github"
    spec:
      user: '{{ requiredEnv "GIT_USER" }}'
      email: '{{ requiredEnv "GIT_EMAIL" }}'
      owner: "elastic"
      repository: "opbeans-node"
      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
      username: '{{ requiredEnv "GIT_USER" }}'
      branch: "main"

sources:
  apm: 
    name: Get latest elastic-apm-node version from npm registry
    kind: npm
    spec: 
      name: elastic-apm-node

targets:
  apm
    name: "Updating apm version"
    sourceid: apm
    scmid: githubConfig
    kind: shell
    spec:
      command: bash -l -c .ci/scripts/update-apm.sh
      environments:
        - name: PATH
          inherit: true
        - name: HOME
          inherit: true
```

Reply:
something like this one, it probably can be improved using a better module for the target.

Reply:
It has an action https://www.updatecli.io/docs/automate/github_action/

Reply:
Okay, thank you. I will take this as a reference. Still need to get familiar with updatecli.

Reply:
@kuisathaverat 

This is how it would look like with updatecli https://github.com/elastic/opbeans-node/pull/179/commits/04e732483fe69ed9f89b42c41b8eb0f2e3f46657.

let me know what you think!

Reply:
> @kuisathaverat or others: What is the value in using `updatecli` here? It looks like this PR got _more_ complex with the usage of updatecli.

From my perspective, it's nice to see instructions defined in a standardized and declarative way. In a wider scope, across many repositories, this could enhance the maintainability and readability.

What I didn't like too much is, defining commands inline did not work as expected. Which is why I had to extract the scripts to a shell file. 

Also, it's unfortunate, that updatecli doesn't seem to return a meaningful exit code. Hence, I was not able to do further logic based on the result and I had to check if there is a git diff.

Another big minus is: The log of the commands are not fully shown. It was hard to debug why my commands did not work in the beginning.

However, I did not check yet, if there are ways to overcome the exit code and log behavior with some args.